### DISCUSSION TITLE
SAP HANA Integration

### DISCUSSION THREAD

**Discussion by tomtom215** [ClickHouse/ClickHouse#43746] [Ideas]
Hi all,

After watching the Clickhouse 22.11 release webinar, Alexey detailed new integrations for Hudi, Delta lake and even mentioned how ClickHouse can accelerate MySQL. Later he made some comments about the challenges of working with Apache and Java related data products so I wanted to raise the idea of a closer integration between SAP HANA and ClickHouse.

Currently, I believe the only "real" method I have found in the documentation is to use the [ClickHouse JDBC bridge,](https://github.com/ClickHouse/clickhouse-jdbc-bridge#performance) which introduces the [Java issue](https://github.com/ClickHouse/clickhouse-jdbc-bridge/issues/164) into the equation. 

Is there a more performant way to make this possible? I listed this as an idea because being a columnar store DB, ClickHouse and SAP HANA usually are listed together, so it would give DB admins a quick way of replicating data sets and running performance & developer experience comparisons.

**Comment by fagzal** [comment]
Ideally there should be a HANA table/db engine for ClickHouse, but I guess there might be licensing issues. I found that the ODBC-bridge is pretty unusable, JDBC bridge is also not recommended... We ended up creating our own Python script, using the native HANA protocol, and configuring it as an Executable table type.
